contracts/HolographInterfaces.sol

```diff
@@ -457,7 +265,18 @@ contract Interfaces is Admin, Initializable {
     _supportedInterfaces[InterfaceType.PA1D][IPA1D.bidSharesForToken.selector] = true;
     _supportedInterfaces[InterfaceType.PA1D][IPA1D.getStorageSlot.selector] = true;
     _supportedInterfaces[InterfaceType.PA1D][IPA1D.getTokenAddress.selector] = true;
+  }
 
+  /**
+   * @notice Used internally to initialize the contract instead of through a constructor
+   * @dev This function is called by the deployer/factory when creating a contract.
+   */
+  function init(bytes memory data) external override returns (bytes4) {
+    require(!_isInitialized(), "HOLOGRAPH: already initialized");
+    address contractAdmin = abi.decode(data, (address));
+    assembly {
+      sstore(_adminSlot, contractAdmin)
+    }
     _setInitialized();
     return IInitializable.init.selector;
```

Comment:
I've noticed this pattern a lot in the holograph contracts. just curious why you return selector here? for use to confirm after contract is initialized

Reply:
Yep, to ensure that the actual function was called and succeeded. Because there is no revert or error if a function is called on a non-deployed/existing contract. This way whenever we call a connected contract, we check for a function selector return to guarantee that the underlying logic exists and was executed successfully. A safeguard against improperly deployed or destroyed contracts since we allow for custom contracts to be implemented by developers.

Reply:
nice thanks for explaining